### PR TITLE
Fix stack alignment computation (#12267)

### PR DESCRIPTION
**Pull request by ghost**
Following issue #12267, this PR:
- relaxes the assertion to only confirm that the allocated stack size is at least equal to the requested size, since alignment may cause more than one extra word to have been allocated.
- fixes the alignment computation of the top of the stack, using the `(size + align - 1) & ~(align - 1)` idiom as found is `round_up()` in `runtime/platform.c` (which can't be used directly as it local to that file).

The previous computation did not round up correctly. If the top of stack would have ended up being 0x...4, it would end up truncated to 0x...0.

### CODE REVIEW

File: runtime/fiber.c [hunk 1]
```diff
@@ -181,8 +181,7 @@ alloc_size_class_stack_noexc(mlsize_t wosize, int cache_bucket, value hval,
 #ifdef DEBUG
   stack->magic = 42;
 #endif
-  CAMLassert(Stack_high(stack) - Stack_base(stack) == wosize ||
-             Stack_high(stack) - Stack_base(stack) == wosize + 1);
+  CAMLassert(Stack_high(stack) - Stack_base(stack) >= wosize);
```
Comment (gasche) [inline runtime/fiber.c:186]: I think that this should either:
- include a comment that explains why the test is weaker than one would naively expect
- or have a more precise test that makes it somewhat self-explanatory; for example, you could check that `high - base` is between `wosize` and `wosize + 15 / sizeof(value)`.

File: Changes [hunk 2]
```diff
@@ -246,6 +246,8 @@ Working version
   raise `Sys_error` in this case.
   (Nicolás Ojeda Bär, review by Gabriel Scherer)
 
+- #12267: Fix stack alignment computation
```
Comment (gasche) [inline Changes:253]: You should include the present PR number and credit yourself, the reporter, and eventually any reviewer.

**Review by gasche — APPROVED** [review]

File: Changes (hunk 2 above)
Comment (ghost) [reply Changes:253]: I'm waiting for the review to happen to amend this.

File: runtime/fiber.c (hunk 1 above)
Comment (ghost) [reply runtime/fiber.c:186]: Done, choosing the first possibility.

**Comment by gasche** [comment]
@dustanddreams could you rebase (there is a Changes conflict) and fixup the history?

**Comment by ghost** [comment]
> @dustanddreams could you rebase (there is a Changes conflict) and fixup the history?

Rebasquashed.

**Comment by gasche** [comment]
Thanks! Merged.

**Comment by dra27** [comment]
Given that there's another fix needed for 5.1 (so presumably another rc?), can I add this to the 5.1 milestone, @Octachron? (and then cherry-pick!)

**Pull request merged**